Documentation/Evolution/StringProcessingAlgorithms.md

```diff
@@ -234,7 +234,7 @@ extension Collection where Element: Equatable {
     /// - Parameter other: A sequence to search for within this collection.
     /// - Returns: `true` if the collection contains the specified sequence,
     /// otherwise `false`.
-    public func contains<S: Sequence>(_ other: S) -> Bool
+    public func contains<C: Collection>(_ other: C) -> Bool
```

Comment:
Would this be `some Collection<Element>`?

Reply:
Right, all of the sequence/collection parameters can change to that. Where there's still a `<R: RegexComponent>` parameter, it's because we need the component's output parameter. Those will change to e.g. `func firstMatch<Output>(of regex: some RegexComponent<Output>) -> Regex<Output>.Match?` when we have the primary associated type.

Reply:
Is the primary associate type tracked anywhere? Do we know when we need to integrate that?

Reply:
It's waiting on stdlib's adoption of lightweight generics. Filed #325 to track this

Reply:
And the `RegexComponent` primary associated type is here: https://github.com/apple/swift-experimental-string-processing/pull/207